src/evo/creditpool.cpp

```diff
@@ -164,7 +164,7 @@ CCreditPool CCreditPoolManager::ConstructCreditPool(const CBlockIndex* const blo
     }
 
     const CBlockIndex* distant_block_index = block_index;
-    for (size_t i = 0; i < CCreditPoolManager::LimitBlocksToTrace; ++i) {
+    for (auto i = 0; i < Params().CreditPoolPeriodBlocks(); ++i) {
```

Comment:
`int` here is just fine

Reply:
> better to auto imo

I'd disagree with your opinion.

There's so many cases, when auto is useful and increase code readability. Because it makes not only shorter to write but also easier to read. For example, iterators for containers, such as std::vector<int>::const_iterator - If type of container is non-trivial even more useful. Also, if there's any lambda function involved, auto's are so useful to assign lambda or pass somewhere.

But in this particular case, `Params().CreditPoolPeriodBlocks();` is not related to type of `auto`, is only thing is matter is a `0` from the initialization; you should read this code such as `auto i = 0`. So, this code compiles without warning, but not because you used `auto`, but because accidentally a type of `Params().CreditPoolPeriodBlocks()` is matched with `int.

In this particular case, `auto` makes code more difficult to read: why there's auto?

yes, it compiles. yes, it works as expected. but there's nothing "better"

Reply:
a1533901a4a847dc9fae602787add7118560ffcd